### PR TITLE
#169257407 Notification for new travel

### PR DESCRIPTION
**Pull request by Mastel22**
#### What does this PR do?
- Create pull request for notification of new travel
#### Description of Task to be completed?
- [X] added the get notification controller
- [X] added socket.io for real-time notifications
- [X]  added the cancel email notification endpoint
- [X] added the email notification service
#### How should this be manually tested?
- Navigate to the root folder where you want to copy this repo
- Run `git clone <link to this repo>
- Run `npm i` to install the dependencies
- Run `npx sequelize-cli db:seed:all` to populate the user table with the needed data
-  Run `npm run watch` to   start the server
- Use postman to:
 1. Sign in using a requester using this url `localhost:3000/api/v1/auth/signin`
2. Create a trip
POST
```
{
	"hotelId": 1,
	"type": "return",
	"leavingFrom": "Nairobi",
	"goingTo": "Kigali",
	"travelDate": "2019-12-02",
	"returnDate": "2019-12-02",
	"reason": "Branch visit in kigali",
	"rooms": [15, 18]
}
```
You should  use a valid email so as to get the notification email

3. Get particular line manager email by signing him in and use his token on route  `localhost:3000/api/v1/notification`

#### Any background context you want to provide?
- If you haven't used socket.io before please read about it before reviewing this PR
#### What are the relevant pivotal tracker stories?
[#169257407](https://www.pivotaltracker.com/story/show/169257407)
#### Screenshots (if appropriate)
image
image

### CODE REVIEW

File: public/templates/emails/emailVerification.pug [hunk 1]
```diff
@@ -36,3 +36,41 @@ button(class='link-btn').
      #[a( href!=`${regenerateLink}` ) Regenerate Link]
 p Yours,
 p bareFoot Team
+html
```
Comment (gildniy) [inline public/templates/emails/emailVerification.pug]: I think this has been duplicated.

File: src/migrations/20191112101055-create-user.js [hunk 2]
```diff
@@ -87,6 +87,11 @@ module.exports = {
     updatedAt: {
       allowNull: false,
       type: Sequelize.DATE
+    },
+    receiveNotification: {
+      type: Sequelize.BOOLEAN,
+      defaultValue: true,
+      allowNull: false
```
Comment (collinewait) [inline src/migrations/20191112101055-create-user.js]: It looks like you updated a migration. You can create a new migration.

File: package.json [hunk 3]
```diff
@@ -46,6 +46,7 @@
     "methods": "^1.1.2",
     "morgan": "^1.9.1",
     "node-mocks-http": "^1.8.0",
+    "nodemailer-mailgun-transport": "^1.4.0",
```
Comment (gildniy) [inline package.json]: I think this package was not used in this branch and may be removed from there.

**Review by gildniy — CHANGES_REQUESTED** [review]
Kindly check on a few feedback I left on this PR.

**Review by b0nbon1 — APPROVED** [review]
I really like how you implemented the `in-app notification` that is by emitting the to the notifications to the user using `socket.io`. Great job
I have reviewed it and I think it's ready to be merged.

**Review by gildniy — DISMISSED** [review]
Looks nice, and I think it's ready to be merged.

**Review by b0nbon1 — DISMISSED** [review]

**Review by nshaphan — DISMISSED** [review]
Great work @Mastel22, This is work is worth the effort.

**Review by b0nbon1 — APPROVED** [review]

**Review by gildniy — APPROVED** [review]
Tested and working. It can be merged, I think.

**Review by collinewait — APPROVED** [review]

**Pull request merged**